### DISCUSSION TITLE
Use VpnHood without UDP

### DISCUSSION THREAD

**Discussion by oSumAtrIX** [vpnhood/VpnHood#46] [Q&A]
Some network is blocking the udp protocol and VpnHood fails to connect because of that.

![IMG-20210924-WA0008](https://user-images.githubusercontent.com/13122796/134631097-8939b8cd-fb24-489f-8e2d-cd8155df21aa.jpg)

Is it possible to establish a connection non the less without udp or tunneling udp over tcp?

**Comment by trudyhood** [comment]
Hi, The diagnose actually tests UDP 8.8.8.8:53 for google, but it is not important, it is just a test and I will remove that test from diagnosing tools.
By the way, VpnHood doesn't use UDP if you use (no UDP) protocol regardless of what the diagnosis does.
Can't you connect with (No UDP) protocol?

![image](https://user-images.githubusercontent.com/72891124/134632773-cea6d71e-23bb-47a5-abf0-4a4b7a2938f3.png)

**Reply by oSumAtrIX** [reply]
No connection can be established regardless having said option enabled or not. The device is connected to [PEAP WIFI](https://it.cornell.edu/wifi/connect-eduroam-android)

**Reply by trudyhood** [reply]
Can you ping one of our public server IPs? Maybe our public IPs are unreachable on your end. Can you send us a complete log.

**Reply by trudyhood** [reply]
Would you let me know your VpnHood Version?

**Reply by oSumAtrIX** [reply]
> Would you let me know your VpnHood Version?

Up to date with the main branch.

> Can you ping one of our public server IPs? Maybe our public IPs are unreachable on your end. Can you send us a complete log.

I am connected to my private homeserver. No logs have been generated on there. The client's logs are the following:
![IMG-20210924-WA0014](https://user-images.githubusercontent.com/13122796/134635382-23a01490-df95-491b-b2c0-7ba5519f9c7c.jpg)

I may note, that connecting from any other regular wifi works without any issues. Sadly logs above were not sufficient for me to figure out what is going on at this specific network

**Reply by trudyhood** [reply]
You can send us a complete log file by pressing "Send Report" so we can investigate more through it.
![image](https://user-images.githubusercontent.com/72891124/134636335-a81dd94d-01b3-4941-a6bc-14c8fd0f1a36.png)

Also please let me us know your internet speed by https://www.speedtest.net/
Make sure you choose a server from U.S to check your speed.

Your issue is important to us and we are going to put time into it to solve that if you are interested to contribute with us to solve this issue.

**Reply by oSumAtrIX** [reply]
I filled the report and sent it.

**Reply by trudyhood** [reply]
I checked your report. It looks like your provider doesn’t let any DNS query over public DNS such as 8.8.8.8. So would you let me know what exactly happens when you connect to VpnHood?
1 ) Can you connect to our public server at all?
2) If yes, what happens? Do you just get connect and then you can't browse any website?
3) Can you browser at-lease one website such as google.com after connecting to our public VpnHood server?

**Comment by oSumAtrIX** [comment]
1) No, when I connect to the default public server, I get the logs seen in https://github.com/vpnhood/VpnHood/discussions/46#discussioncomment-1378403.
2) HttpTest succeeds, but UDP connection fails
3) No connection can be established as seen in 1)

**Reply by trudyhood** [reply]
Can you navigate to google via IP:

https://172.217.4.174/

It is expected to see the following error page, Please connect to our public server and then check the link. It is expected to see the following error. Let me know the result.
![image](https://user-images.githubusercontent.com/72891124/135689364-c28598b8-db91-4dcc-9ddc-8b6f14921c43.png)
